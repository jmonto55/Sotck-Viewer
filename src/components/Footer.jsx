const Footer = () => (
  <div className="w-full pb-5">
    <footer className="bg-neutral-900 dark:bg-slate-100">
      <p className="text-base text-center text-white mb-2 dark:text-neutral-900">Â© 2023 Jose Montoya. All rights reserved.</p>
      <p className="text-sm text-center text-white dark:text-neutral-900">Built withðŸ’™in Medellin, CO</p>
    </footer>
  </div>
);

export default Footer;
