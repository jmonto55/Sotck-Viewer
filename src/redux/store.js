import { configureStore } from '@reduxjs/toolkit';
import marketsReducer from './markets/marketsSlice';

const store = configureStore({
  reducer: {
    markets: marketsReducer,
  },
});

export default store;
