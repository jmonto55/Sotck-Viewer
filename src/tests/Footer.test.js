import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Footer from '../components/Footer';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Test suite for the Footer component', () => {
  it('Should render a Snapshot of Footer component', () => {
    const footer = render(
      <Provider store={store}>
        <Footer />
      </Provider>,
    );
    expect(footer).toMatchSnapshot();
  });

  test('Should test if the text "Built withðŸ’™in Medellin, CO" is inside the Footer', () => {
    render(
      <Provider store={store}>
        <Footer />
      </Provider>,
    );
    expect(screen.getByText('Built withðŸ’™in Medellin, CO')).toBeInTheDocument();
  });
});
