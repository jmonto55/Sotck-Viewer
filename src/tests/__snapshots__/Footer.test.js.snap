// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Footer component Should render a Snapshot of Footer component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <footer
          class="bg-neutral-900"
        >
          <p
            class="text-base text-center text-white mb-2"
          >
            Â© 2023 Jose Montoya. All rights reserved.
          </p>
          <p
            class="text-sm text-center text-white"
          >
            Built withðŸ’™in Medellin, CO
          </p>
          <div
            class="flex justify-evenly"
          >
            <img
              alt="github"
              class="opacity-10"
              src="github.svg"
            />
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <footer
        class="bg-neutral-900"
      >
        <p
          class="text-base text-center text-white mb-2"
        >
          Â© 2023 Jose Montoya. All rights reserved.
        </p>
        <p
          class="text-sm text-center text-white"
        >
          Built withðŸ’™in Medellin, CO
        </p>
        <div
          class="flex justify-evenly"
        >
          <img
            alt="github"
            class="opacity-10"
            src="github.svg"
          />
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
